File: src/setup/Question.php
<?php

namespace pocketcloud\cloud\setup;

use Closure;

readonly class Question {

    public function __construct(
        private string $id,
        private string $question,
        private bool $canSkipped,
        private array $possibleAnswers,
        private ?string $defaultValueMessage,
        private mixed $defaultValue,
        private ?string $recommendation,
        private Closure $parser,
        private ?Closure $resultHandler
    ) {}

    public function getId(): string {
        return $this->id;
    }

    public function getQuestion(): string {
        return $this->question;
    }

    public function isCanSkipped(): bool {
        return $this->canSkipped;
    }

    public function getPossibleAnswers(): array {
        return $this->possibleAnswers;
    }

    public function getDefaultValueMessage(): ?string {
        return $this->defaultValueMessage;
    }

    public function getDefaultValue(): mixed {
        return $this->defaultValue;
    }

    public function getRecommendation(): ?string {
        return $this->recommendation;
    }

    public function getParser(): Closure {
        return $this->parser;
    }

    public function getResultHandler(): ?Closure {
        return $this->resultHandler;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/setup/Question.java
package de.pocketcloud.cloud.setup;

import lombok.Getter;

import java.util.Collections;
import java.util.List;
import java.util.function.Consumer;

@Getter
public final class Question<T> {

    private final String id;
    private final String question;
    private final boolean canSkipped;
    private final List<String> possibleAnswers;
    private final String defaultValueMessage;
    private final T defaultValue;
    private final String recommendation;
    private final QuestionParser<T> parser;
    private final Consumer<T> resultHandler;

    Question(
            String id,
            String question,
            boolean canSkipped,
            List<String> possibleAnswers,
            String defaultValueMessage,
            T defaultValue,
            String recommendation,
            QuestionParser<T> parser,
            Consumer<T> resultHandler
    ) {
        this.id = id;
        this.question = question;
        this.canSkipped = canSkipped;
        this.possibleAnswers = possibleAnswers == null ? Collections.emptyList() : possibleAnswers;
        this.defaultValueMessage = defaultValueMessage;
        this.defaultValue = defaultValue;
        this.recommendation = recommendation;
        this.parser = parser;
        this.resultHandler = resultHandler;
    }
}